xlog/packagelogger.go

```diff
@@ -124,35 +124,53 @@ func (p *PackageLogger) Error(entries ...interface{}) {
 
 // Warningf is implementation for stdlib compatibility
 func (p *PackageLogger) Warningf(format string, args ...interface{}) {
+	if p.level < WARNING {
```

Comment:
Sorry i am not getting why is this block needed?  That block is equivalent to `p.Level==Level.ERROR or p.Level==LEVEL.CRITICAL` than return without logging. But is not p.Logf(**WARNING**,...) supposed to handle that already (e.g. for levels lower that WARNING it should not log the WARNING messages)?

Reply:
@haykbaluyan `p.Logf` passes the call to `p.internalLog` with  `Sprintf`.
And this is unnecessary perf impact, if the level is not enabled.